app/addons/auth/actions.js

```diff
@@ -28,88 +28,130 @@ var errorHandler = function (xhr, type, msg) {
-      FauxtonAPI.dispatch({ type: ActionTypes.AUTH_CLEAR_CHANGE_PWD_FIELDS });
+    hidePasswordModal();
+    onSuccess(username, password);
+  }, () => {
+    FauxtonAPI.addNotification({
+      msg: 'Your password is incorrect.',
+      type: 'error',
+      clear: true
     });
-
-    promise.fail(errorHandler);
-  },
-
-  updateChangePasswordField: function (value) {
     FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_UPDATE_CHANGE_PWD_FIELD,
-      value: value
+      type: ActionTypes.AUTH_CREDS_INVALID,
+      options: { username: username, password: password }
     });
-  },
+  });
+}
 
-  updateChangePasswordConfirmField: function (value) {
-    FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_UPDATE_CHANGE_PWD_CONFIRM_FIELD,
-      value: value
-    });
-  },
-
-  createAdmin: function (username, password, loginAfter) {
-    var nodes = nodesStore.getNodes();
-    var promise = FauxtonAPI.session.createAdmin(username, password, loginAfter, nodes[0].node);
-
-    promise.then(function () {
-      FauxtonAPI.addNotification({ msg: FauxtonAPI.session.messages.adminCreated });
-      if (loginAfter) {
-        FauxtonAPI.navigate('/');
-      } else {
-        FauxtonAPI.dispatch({ type: ActionTypes.AUTH_CLEAR_CREATE_ADMIN_FIELDS });
-      }
-    });
+function updateCreateAdminUsername (value) {
+  FauxtonAPI.dispatch({
+    type: ActionTypes.AUTH_UPDATE_CREATE_ADMIN_USERNAME_FIELD,
+    value: value
+  });
+}
 
-    promise.fail(function (xhr, type, msg) {
-      msg = xhr;
-      if (arguments.length === 3) {
-        msg = xhr.responseJSON.reason;
-      }
-      errorHandler(FauxtonAPI.session.messages.adminCreationFailedPrefix + ' ' + msg);
-    });
-  },
+function updateCreateAdminPassword (value) {
+  FauxtonAPI.dispatch({
+    type: ActionTypes.AUTH_UPDATE_CREATE_ADMIN_PWD_FIELD,
+    value: value
+  });
+}
 
-  updateCreateAdminUsername: function (value) {
-    FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_UPDATE_CREATE_ADMIN_USERNAME_FIELD,
-      value: value
-    });
-  },
+function selectPage (page) {
+  FauxtonAPI.dispatch({
+    type: ActionTypes.AUTH_SELECT_PAGE,
+    page: page
+  });
+}
 
-  updateCreateAdminPassword: function (value) {
-    FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_UPDATE_CREATE_ADMIN_PWD_FIELD,
-      value: value
-    });
-  },
+function showPasswordModal () {
+  FauxtonAPI.dispatch({ type: ActionTypes.AUTH_SHOW_PASSWORD_MODAL });
+}
+
+function hidePasswordModal () {
+  FauxtonAPI.dispatch({ type: ActionTypes.AUTH_HIDE_PASSWORD_MODAL });
+}
 
-  selectPage: function (page) {
-    FauxtonAPI.dispatch({
-      type: ActionTypes.AUTH_SELECT_PAGE,
-      page: page
-    });
-  }
 
+export default {
+  login,
+  changePassword,
+  updateChangePasswordField,
+  updateChangePasswordConfirmField,
+  createAdmin,
+  authenticate,
+  updateCreateAdminUsername,
+  updateCreateAdminPassword,
+  selectPage,
+  showPasswordModal,
+  hidePasswordModal
```

Comment:
is this part of the replication work or a refactor? did something change, except the function type?